core/app/api/makeswift/[...makeswift]/route.ts

```diff
@@ -22,29 +19,67 @@ const defaultVariants: Font['variants'] = [
   },
 ];
 
-const handler = MakeswiftApiHandler(process.env.MAKESWIFT_SITE_API_KEY, {
-  runtime,
-  apiOrigin: process.env.MAKESWIFT_API_ORIGIN,
-  appOrigin: process.env.MAKESWIFT_APP_ORIGIN,
-  getFonts() {
-    return [
-      {
-        family: 'var(--font-family-inter)',
-        label: 'Inter',
-        variants: defaultVariants,
-      },
-      {
-        family: 'var(--font-family-dm-serif-text)',
-        label: 'DM Serif Text',
-        variants: [{ weight: '400', style: 'normal' }],
-      },
-      {
-        family: 'var(--font-family-roboto-mono)',
-        label: 'Roboto Mono',
-        variants: defaultVariants,
-      },
-    ];
+const getHandler = (apiKey: string, apiOrigin: string | undefined, appOrigin: string | undefined) => {
+  return MakeswiftApiHandler(apiKey, {
+    apiOrigin,
+    appOrigin,
+    runtime,
+    getFonts() {
+      return [
+        {
+          family: 'var(--font-family-inter)',
+          label: 'Inter',
+          variants: defaultVariants,
+        },
+        {
+          family: 'var(--font-family-dm-serif-text)',
+          label: 'DM Serif Text',
+          variants: [{ weight: '400', style: 'normal' }],
+        },
+        {
+          family: 'var(--font-family-roboto-mono)',
+          label: 'Roboto Mono',
+          variants: defaultVariants,
+        },
+      ];
+    },
+  });
+};
+
+export async function GET(
+  request: NextRequest,
+  {
+    params,
+  }: {
+    params: { makeswift: string };
+  },
+) {
+  const apiKey = process.env.MAKESWIFT_SITE_API_KEY;
+  const apiOrigin = process.env.MAKESWIFT_API_ORIGIN;
+  const appOrigin = process.env.MAKESWIFT_APP_ORIGIN;
+
+  if (!apiKey) {
+    throw new Error('MAKESWIFT_SITE_API_KEY is required');
+  }
+
+  return getHandler(apiKey, apiOrigin, appOrigin)(request, { params });
```

Comment:
I'd factor out the env vars extraction into its own function so that this and the POST handler call can become just

```typescript
  return getHandler(makeswiftEnv())(request, { params });
```

Reply:
E.g.

```typescript
type MakeswiftEnv = { apiKey: string, apiOrigin: string | undefined, appOrigin: string | undefined }
const makeswiftEnv = (): MakeswiftEnv => ...

const getHandler = ({apiKey, apiOrigin, appOrigin}: MakeswiftEnv) => ...

export async function GET(
  request: NextRequest,
  {
    params,
  }: {
    params: { makeswift: string };
  },
) {
  return getHandler(makeswiftEnv())(request, { params });
}
```